includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -336,24 +336,29 @@ protected function get_products_by_order_ids( $order_ids ) {
 	}
 
 	/**
-	 * Get customer data from order IDs.
+	 * Get customer data from Order data.
 	 *
-	 * @param array $orders Array of orders.
+	 * @param array $orders Array of orders data.
 	 * @return array
 	 */
 	protected function get_customers_by_orders( $orders ) {
 		global $wpdb;
+
 		$customer_lookup_table = $wpdb->prefix . 'wc_customer_lookup';
+		$customer_ids          = array();
 
-		$customer_ids = array();
 		foreach ( $orders as $order ) {
 			if ( $order['customer_id'] ) {
 				$customer_ids[] = $order['customer_id'];
 			}
 		}
-		$customer_ids = implode( ',', $customer_ids );
 
-		$customers = $wpdb->get_results(
+		if ( empty( $customer_ids ) ) {
+			return array();
+		}
+
+		$customer_ids = implode( ',', $customer_ids );
```

Comment:
You need to use array_map here and apply sanitization to the customer_ids - absint would suffice, because it's being appended to the SQL query below without escaping.

Reply:
Added in 4e8eccd.